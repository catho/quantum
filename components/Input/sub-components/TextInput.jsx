import PropTypes from 'prop-types';
import styled from 'styled-components';
import { spacing, colors } from '../../shared/theme';
import { shadow } from '../../shared';

const TextInput = styled.input`
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  letter-spacing: 0.2px;
  outline: none;
  transition: all 0.2s ease-in-out;
  width: 100%;

  &::-webkit-inner-spin-button,
  &::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  &::-webkit-calendar-picker-indicator,
  &::-webkit-search-cancel-button {
    display: none;
  }

  ${({
    defaultValue,
    error,
    hasRightIcon,
    hasLeftIcon,
    placeholder,
    theme,
    value,
  }) => {
    const {
      colors: {
        primary: { 700: primaryColor },
        error: { 100: error100, 700: error700 },
        neutral: {
          0: neutral0,
          300: neutral300,
          500: neutral500,
          700: neutral700,
        },
      },
      spacing: { xsmall, small, xxlarge },
    } = theme;

    const mainColor = error ? error700 : primaryColor;

    return `
      background-color: ${neutral0};
      border: 2px solid ${neutral500};
      color: ${neutral700};
      margin-top: ${xsmall}px;
      padding: ${xsmall}px ${small}px;

      ${hasRightIcon ? `padding-right: ${xxlarge}px;` : ''}
      ${hasLeftIcon ? `padding-left: ${xxlarge}px;` : ''}
      ${error ? `border-color: ${mainColor};` : ''}
      ${placeholder && !defaultValue && !value ? `color: ${neutral500};` : ''}

      :hover, :focus {
        border-color: ${mainColor};
        ${shadow(5, mainColor)({ theme })}
      }

      &[disabled] {
        background-color: ${neutral300};
        border-color: ${neutral500};
        box-shadow: none;
        color: ${neutral500};
        cursor: not-allowed;
      }

      :-webkit-autofill {
        box-shadow: 0 0 0 1000px ${error100} inset;
      }
    `;
  }}
`;

TextInput.displayName = 'TextInput';

TextInput.propTypes = {
  theme: PropTypes.shape({
    spacing: PropTypes.object,
    colors: PropTypes.object,
  }),
};

TextInput.defaultProps = {
  theme: { spacing, colors },
};

export default TextInput;
