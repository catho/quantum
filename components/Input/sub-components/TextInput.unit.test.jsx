import { render } from '@testing-library/react';
import TextInput from './TextInput';

describe('Input component', () => {
  it('should match snapshots', () => {
    const INPUTS = [
      <TextInput />,
      <TextInput skin="dark" />,
      <TextInput error="Error message" />,
      <TextInput skin="dark" error="Error message" />,
      <TextInput disabled />,
      <TextInput skin="dark" disabled />,
      <TextInput placeholder="Placeholder text" />,
      <TextInput skin="dark" placeholder="Placeholder text" />,
      <TextInput value="Value text" readOnly />,
      <TextInput skin="dark" value="Value text" readOnly />,
      <TextInput defaultValue="Value text" />,
      <TextInput skin="dark" defaultValue="Value text" />,
      <TextInput placeholder="Placeholder text" value="Value text" readOnly />,
      <TextInput
        skin="dark"
        placeholder="Placeholder text"
        value="Value text"
        readOnly
      />,
      <TextInput error="Error message" disabled />,
      <TextInput skin="dark" error="Error message" disabled />,
    ];

    INPUTS.forEach((input) => {
      const { container } = render(input);
      expect(container).toMatchSnapshot();
    });
  });
});
