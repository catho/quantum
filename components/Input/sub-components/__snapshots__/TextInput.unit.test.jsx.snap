// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should match snapshots 1`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 2`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  border-color: #e91b0c;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #e91b0c;
  box-shadow: 0px 3px 1px -2px rgba(233,27,12,0.2),0px 2px 2px 0px rgba(233,27,12,0.14),0px 1px 5px 0px rgba(233,27,12,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 3`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
  disabled={true}
/>
`;

exports[`Input component should match snapshots 4`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
/>
`;

exports[`Input component should match snapshots 5`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  color: #999999;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
  placeholder="Placeholder text"
/>
`;

exports[`Input component should match snapshots 6`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
  value="Value text"
/>
`;

exports[`Input component should match snapshots 7`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
  defaultValue="Value text"
/>
`;

exports[`Input component should match snapshots 8`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(18,80,196,0.2),0px 2px 2px 0px rgba(18,80,196,0.14),0px 1px 5px 0px rgba(18,80,196,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
  placeholder="Placeholder text"
  value="Value text"
/>
`;

exports[`Input component should match snapshots 9`] = `
.c0 {
  -webkit-appearance: none;
  border-radius: 4px;
  box-sizing: border-box;
  box-sizing: border-box;
  font-size: initial;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  outline: none;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  background-color: #ffffff;
  border: 2px solid #999999;
  color: #424242;
  margin-top: 8px;
  padding: 8px 16px;
  border-color: #e91b0c;
}

.c0::-webkit-inner-spin-button,
.c0::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c0::-webkit-calendar-picker-indicator,
.c0::-webkit-search-cancel-button {
  display: none;
}

.c0::-ms-clear {
  display: none;
}

.c0:hover,
.c0:focus {
  border-color: #e91b0c;
  box-shadow: 0px 3px 1px -2px rgba(233,27,12,0.2),0px 2px 2px 0px rgba(233,27,12,0.14),0px 1px 5px 0px rgba(233,27,12,0.12);
}

.c0[disabled] {
  background-color: #f2f2f2;
  border-color: #999999;
  box-shadow: none;
  color: #999999;
  cursor: not-allowed;
}

.c0:-webkit-autofill {
  box-shadow: 0 0 0 1000px #ffe5e9 inset;
}

<input
  className="c0"
  disabled={true}
/>
`;
